include/mppic/parameters_handler.hpp

```diff
@@ -0,0 +1,138 @@
+// Copyright 2022 FastSense, Samsung Research
+
+#pragma once
+
+#include <vector>
+#include <functional>
+#include <type_traits>
+
+#include "rclcpp/rclcpp.hpp"
+#include "rclcpp_lifecycle/lifecycle_node.hpp"
+#include "nav2_util/node_utils.hpp"
+
+namespace mppi
+{
+
+enum class ParameterType
+{
+  Dynamic,
+  Static
+};
+
+class ParametersHandler
+{
+public:
+  using get_param_func_t = void (const rclcpp::Parameter & param);
+  using post_callback_t = void ();
+
+  ParametersHandler() = default;
+  ParametersHandler(const rclcpp_lifecycle::LifecycleNode::WeakPtr & parent)
+  {
+    node_ = parent;
+    auto node = node_.lock();
+    logger_ = node->get_logger();
+  }
+
+  void start()
+  {
+    auto node = node_.lock();
+    on_set_param_handler_ = node->add_on_set_parameters_callback(
+      std::bind(
+        &ParametersHandler::dynamicParametersCallback, this, std::placeholders::_1));
+  }
+
+  rcl_interfaces::msg::SetParametersResult dynamicParametersCallback(
```

Comment:
This design is going to be nearly impossible to lock these resources so that we don't have issues where variables are being written over by dynamic parameter updates during a control cycle that is trying to access it (giving garbage memory)